File: models/Student.ts
import mongoose, { Schema, type Document } from "mongoose"

export interface IStudent extends Document {
  fullName: string
  groupId: mongoose.Types.ObjectId
  createdAt: Date
}

const StudentSchema = new Schema<IStudent>({
  fullName: {
    type: String,
    required: true,
  },
  groupId: {
    type: Schema.Types.ObjectId,
    ref: "Group",
    required: true,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
})

export default mongoose.models.Student || mongoose.model<IStudent>("Student", StudentSchema)

File: flutter_app/lib/models/student.dart
class Student {
  final String id;
  final String fullName;
  final String groupId;
  final DateTime createdAt;

  Student({
    required this.id,
    required this.fullName,
    required this.groupId,
    required this.createdAt,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['_id'] ?? json['id'] ?? '',
      fullName: json['fullName'] ?? '',
      groupId: json['groupId'] ?? '',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'fullName': fullName,
      'groupId': groupId,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
